src/frontend/OverlayManager/Overlay/NavBarOverlayWrapper/index.tsx

```diff
@@ -3,13 +3,51 @@ import styles from './index.module.scss'
 import { t } from 'i18next'
 import { NavBarOverlay, NavItem, Images, SubLink } from '@hyperplay/ui'
 import { Link, useLocation } from 'react-router-dom'
+import { BrowserGameProps } from 'frontend/OverlayManager/types'
+import { useQuery } from '@tanstack/react-query'
+import classNames from 'classnames'
 
-export function NavBarOverlayWrapper() {
+export function NavBarOverlayWrapper({ appName, runner }: BrowserGameProps) {
   const location = useLocation()
-  const { pathname } = location
+  const { pathname, search } = location
+  const searchParams = new URLSearchParams(search)
+  const urlQueryParam = searchParams.get('url')
   const [collapsed, setCollapsed] = useState(false)
   const comingSoonText = t('overlay.comingSoon', 'Coming Soon')
   const [mmCollapsed, mmSetCollapsed] = useState(false)
+  const [mktCollapsed, setMktCollapsed] = useState(false)
+
+  const { data } = useQuery({
+    queryKey: ['getGameInfo', appName],
+    queryFn: async () => {
+      return window.api.getGameInfo(appName, runner)
+    }
+  })
+
+  const marketplaceEnabled =
```

Comment:
Maybe this could be a helper method on a utils file or even on the same file to make it more readable.

Reply:
Refactored to subcomponents and broke up the && to two separate more descriptive named variables